packages/contracts-bedrock/src/L2/SuperchainERC20Bridge.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+// Interfaces
+import { ISemver } from "src/universal/interfaces/ISemver.sol";
+import { ISuperchainERC20Bridge } from "src/L2/interfaces/ISuperchainERC20Bridge.sol";
+import { IMintableAndBurnableERC20 } from "src/L2/interfaces/IMintableAndBurnableERC20.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/interfaces/IL2ToL2CrossDomainMessenger.sol";
+
+/// @custom:proxied true
+/// @custom:predeploy 0x4200000000000000000000000000000000000028
+/// @title SuperchainERC20Bridge
+/// @notice The SuperchainERC20Bridge allows for the bridging of ERC20 tokens to make them fungible across the
+///         Superchain. It builds on top of the L2ToL2CrossDomainMessenger for both replay protection and domain
+///         binding.
+contract SuperchainERC20Bridge is ISuperchainERC20Bridge, ISemver {
+    /// @notice Address of the L2ToL2CrossDomainMessenger Predeploy.
+    address internal constant MESSENGER = Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER;
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
```

Comment:
Question: Should we maintain the same version for the redesign?

Reply:
The same version as what? I put this one because it is the first iteration of this bridge contract, but we can change it